src/libraries/System.Text.RegularExpressions/src/System/Text/RegularExpressions/RegexCharClass.cs

```diff
@@ -873,14 +936,68 @@ static bool MayOverlapByEnumeration(string set1, string set2)
         }
 
         /// <summary>Gets whether we can iterate through the set list pairs in order to completely enumerate the set's contents.</summary>
-        internal static bool CanEasilyEnumerateSetContents(string set) =>
+        private static bool CanEasilyEnumerateSetContents(string set) =>
             set.Length > SetStartIndex &&
             set[SetLengthIndex] > 0 &&
             set[SetLengthIndex] % 2 == 0 &&
             set[CategoryLengthIndex] == 0 &&
-            !IsNegated(set) &&
             !IsSubtraction(set);
 
+        /// <summary>Provides results from <see cref="Analyze"/>.</summary>
+        internal struct CharClassAnalysisResults
+        {
+            /// <summary>true if we know for sure that the set contains only ASCII values; otherwise, false.</summary>
+            public bool ContainsOnlyAscii;
+            /// <summary>true if we know for sure that the set doesn't contain any ASCII values; otherwise, false.</summary>
+            public bool ContainsNoAscii;
+            /// <summary>true if we know for sure that all ASCII values are in the set; otherwise, false.</summary>
+            public bool AllAsciiContained;
+            /// <summary>true if we know for sure that all non-ASCII values are in the set; otherwise, false.</summary>
+            public bool AllNonAsciiContained;
+        }
```

Comment:
Wouldn't this be better as a bitmask? Saves allocations in `Analyze()` below.

Reply:
This is a struct, so there's no heap allocation, and as written it's 4 bytes, so no worse than an Int32.  I could make it a bit flags to get it to 1 byte, but the difference doesn't seem particularly meaningful, and in the near future we might want to augment this further, such as with a min/max value that can be used for comparisons instead of 128.

Reply:
I need to learn more about C#. :)

Reply:
:smile: